### ISSUE TITLE
Investigate the viability of "background matching" technique

### ISSUE THREAD

**Issue by tms-epcc** [lsst-uk/sky-estimation-WP3.7#45]
From Dec'22 quarterly update
- Investigate the viability of a second-order solution proposed by DM called "background matching" for possible use in full	   survey.
- this is a stretch goal for Phase B 
- not a first order priority

**Comment by tms-epcc** [comment]
10/MAY/23
- met DM on 08/MAY/23 to discuss this

**Comment by tms-epcc** [comment]
30/MAY/23
- working on other priorities so no progress

**Comment by tms-epcc** [comment]
15/JUN/23
- no progress due to other priorities

**Comment by tms-epcc** [comment]
04/JUL/23
- expect to start discussions with DM team on implementation
- automation a key consideration for metrics

**Comment by tms-epcc** [comment]
20/JUL/23
- continuing to investigate viability
- on  hold 
- this is a stretch goal

**Comment by tms-epcc** [comment]
11/APR/24
- This was originally a stretch goal but further investigations has taken place
- Aaron made synthetic images and assessed what is meant to happen and why. 
- Able to get same sky across all coadds
- DM have some existing software that Aaron may finish if appropriate

**Comment by tms-epcc** [comment]
02/MAY/24
- progressing this 
- there is a version in the LSST github.
- asked Robert Lupton about  status of this but no response yet
- broadening strategy by considering other options 
- challenge now is image quality which has a lot of factors

**Comment by tms-epcc** [comment]
23/MAY/24
- recvd confirmation that the DM Github package is doing background matching
- found that package had been tried on HSC pipeline but DM team found it did not work as they wanted so it has been set aside by DM
- One option is therefore for @aemerywatkins to get involved and take this package further 
- For the moment will investigate this on HSC and will talk to Robert and Yusra on why exactly the package was shelved.

**Comment by tms-epcc** [comment]
17/JUN/24
- got technique working on own images 
- indicates this is viable 
- presented to Lee and others in DM
- will discuss more with Lee 
- will then set up ticket to revise the code and then test on HSC images

**Comment by tms-epcc** [comment]
09/JUL/24
- See DM JIRA ticket - https://rubinobs.atlassian.net/browse/DM-44889
- existing code in LSST Github - matchbackgrounds.py written some time ago
- @aemerywatkins getting this working with Gen3 
- mostly done - code executes apart from last part when writing images to disk

**Comment by tms-epcc** [comment]
23/JUL/24
- code is functioning 
- run on series of images, output was sensible
- working with Lee on how to optimise and meet DM's code standards
- this approach is still the most attractive since it means people can reprocess images themselves rather than having to supercomputing time to do this. 
- still to try running this at scale and obviously still to convince DM that this is the most appropriate technique
- so currently ironing out various details

**Comment by tms-epcc** [comment]
06/AUG/24
- Robert Lupton taking great interest and providing ideas
- Suggested try the technique on old M31 galaxy images since if it works on these it should work on many others
- So currently working on this

**Comment by tms-epcc** [comment]
24/SEP/24
- discussing with Lee that Robert is always asking about the project
- gave the M31 processing a try but pipeline crashing for various reasons so have put on hold for now
- there are now DE images of the Triffid nebula that have gone further on the pipeline so may try on that 
- for now though Aaron continuing with development and working on the issues raised by Lee previously

**Comment by tms-epcc** [comment]
22/OCT/24
- image patterns are staying in the image w/o sky subtraction (see https://rubinobs.atlassian.net/browse/DM-46693)
   - currently under review 
   - one further item to complete before merge
- Lee and Aaron met with Jim Bosch to discuss next steps
   - investigating whether depending on technique,  crashes will occur in certain parts of sky

**Comment by tms-epcc** [comment]
12/NOV/24
- skycor modification needs review before merge
- considering Jim Bosch's ideas as well as crashes this includes concerns about contiguity
- will visit Princeton in Dec'24 and discuss further with Jim in-person

**Comment by tms-epcc** [comment]
28/NOV/24
- waiting on skycor change merge still; need Lee to look at this
- contiguity issue solved but there are some lingering onces
- will still discuss with Jim Bosch

**Comment by tms-epcc** [comment]
14/JAN/25
- skycor merge successful
- a bug discovered in the warping code and being investigated. This is preventing final implementation of the warping code.
- Still to discuss with Jim Bosch.

**Comment by tms-epcc** [comment]
31/JAN/25
- the possible bug was not a bug, it is performing as expected. The issue was the test image.  Subsequently tried with other test images.
- matched background will either have to be more complex or will have to determine how to remove the background
- with Lee will investigate the best way forward

**Comment by tms-epcc** [comment]
14/FEB/25
- testing possible fixes
- expect fix is to fit a function that is background + a correction factor
- now testing this

**Comment by tms-epcc** [comment]
28/FEB/25
- discussions with Robert Lupton o/g who is trying to identify where the problem pattern is from
- a solution has been proposed but not clear yet how this will work
- Lee and Robert on leave so will pick up again in a week's time

**Comment by tms-epcc** [comment]
21/MAR/25
- Aaron still working with Lee on this

**Comment by tms-epcc** [comment]
11/APR/25
- Aaron will tidy match background code and push to main branch where it will get more attention from rest of DM team eg Jim Bausch.  Want to agree a plan with DM on how to get this integrated and progressed.

**Comment by tms-epcc** [comment]
02/MAY/25
- still tidying, iterating this with Lee 
- Aaron and Lee will test the background matching code on ComCam code

**Comment by tms-epcc** [comment]
16/MAY/25
- tested successfully on ComCam
- Robert still +ve that this work
- Lee trying to get skycor working on LSSTcam
- Jim Bausch altering the other correction software
- in the meantime, Aaron will take more active role in science unit and sources of scatter light
- potentially can use Eli Rykoff data product to deal with warping

**Comment by tms-epcc** [comment]
06/JUN/25
- PR submitted
- Jim Bosch will be the reviewer
- need DM input to progress, the PR will provoke this

**Comment by tms-epcc** [comment]
27/JUN/25
- waiting on Jim's input on the PR

**Comment by tms-epcc** [comment]
15/JUL/25
- still waiting on Jim
- not urgent, would be good if ready by Dec'25

**Comment by tms-epcc** [comment]
05/AUG/25
- no news

**Comment by tms-epcc** [comment]
30/SEP/25
- no news probably due to last phase of commissioning DM have been quiet
- expecting once telescope in obs that this will pick up

**Comment by tms-epcc** [comment]
21/OCT/25
- Lee says Jim Bausch is looking at the PR
- Aaron will be meeting weekly with Robert Lupton's optimistic aim being to have it ready by DP2

**Comment by tms-epcc** [comment]
11/NOV/25
- will ensure software runs on LSST Cam before
- meeting Robert weekly from next week
- will meet Lee and Johnny E to discuss in advance

**Comment by tms-epcc** [comment]
25/NOV/25
- merged changes into main but not yet production-ready
- expecting others to make branches for further development
- Aaron's activity and hence the ticket on this are closed
